pkg/testing/utils/utils.go

```diff
@@ -154,6 +154,35 @@ func BuildProc(comm string, uid, gid uint32) Process {
 	}
 }
 
+// NormalizeParentTid normalizes the parent TID to the first thread (main thread)
+// of the process by reading /proc/$tid/task/ directory. This allows comparing
+// parent TIDs that might be different threads of the same process.
+func NormalizeParentTid(p *Process) {
+	if p.Parent.Tid == 0 {
+		return
+	}
+
+	taskDir := fmt.Sprintf("/proc/%d/task", p.Parent.Tid)
```

Comment:
Should this use the PID instead?

Reply:
I dunno, just followed Alban's [lead](https://github.com/inspektor-gadget/inspektor-gadget/issues/5068#issue-3612709641):

> This could be done by checking if the two /proc/$PARENT_THREAD_ID/task/ have the same first thread in the list.